lib/between_meals/repo.rb

```diff
@@ -35,6 +35,28 @@ def initialize(repo_path, logger)
 
     def self.get(type, repo_path, logger)
       case type
+      when 'auto'
+        logger.info('Trying to detect your repo type')
+        require 'between_meals/repo/git'
+        require 'between_meals/repo/hg'
+        require 'between_meals/repo/svn'
+        [
+          BetweenMeals::Repo::Git,
+          BetweenMeals::Repo::Hg,
+          BetweenMeals::Repo::Svn,
+        ].each do |klass|
+          begin
+            r = klass.new(repo_path, logger)
+            if r.exists?
+              logger.info("Repo found to be #{r.bin}")
```

Comment:
not in. `bin` is used in lieu of `type` which is not a field, more friendly/less confusing than i.e. `BetweenMeals::Repo::Git`. I can change this to `klass.to_s.split('::').last`